tests/WCPAY_UnitTestCase.php

```diff
@@ -48,16 +48,19 @@ protected function mock_wcpay_request( string $request_class, int $total_api_cal
 			// No expectation for calls, return here.
 			return;
 		}
+		// Since setMethodsExcept is deprecated, this is the only alternative I came upon.
+		$methods_to_mock = array_diff( get_class_methods( $request_class ), [ 'set_hook_args', 'apply_hook' ] );
 
 		$request = $this->getMockBuilder( $request_class )
 			->setConstructorArgs( [ $api_client_mock, $http_mock, $request_class_constructor_id ] )
+			->onlyMethods( $methods_to_mock )  // Mock all methods except set_hook_args and apply_hook to accommodate filter args when apply_filters is called.
 			->getMock();
 
 		$api_client_mock->expects( $this->exactly( $total_api_calls ) )
 			->method( 'send_request' )
 			->with(
 				$this->callback(
-					// With filters there is a chance that mock will be changed. With this code we are sure that it belongs to same class.
+							// With filters there is a chance that mock will be changed. With this code we are sure that it belongs to same class.
```

Comment:
Something happened with tabs here. Great approach otherwise!

Reply:
Thank you! On another note, the issue with tabbing here: https://github.com/Automattic/woocommerce-payments/pull/7314/commits/e39ac1fe049de9ff790af034762fead702a8fbe9